test/common/config/utility_test.cc

```diff
@@ -81,73 +81,9 @@ TEST(UtilityTest, TranslateApiConfigSource) {
-  tag_extractors = Utility::createTagExtractors(bootstrap);
-  EXPECT_EQ(0, tag_extractors.size());
-
-  // Default extractors explicitly tuned on.
-  stats_config.mutable_use_all_default_tags()->set_value(true);
-  tag_extractors = Utility::createTagExtractors(bootstrap);
-  EXPECT_EQ(tag_names.name_regex_pairs_.size(), tag_extractors.size());
-
-  // Create a duplicate name by adding a default name with all the defaults enabled.
-  auto& custom_tag_extractor = *stats_config.mutable_stats_tags()->Add();
-  custom_tag_extractor.set_tag_name(tag_names.CLUSTER_NAME);
-  EXPECT_THROW_WITH_MESSAGE(Utility::createTagExtractors(bootstrap), EnvoyException,
-                            fmt::format("Tag name '{}' specified twice.", tag_names.CLUSTER_NAME));
-
-  // Remove the defaults and ensure the manually added default gets the correct regex.
-  stats_config.mutable_use_all_default_tags()->set_value(false);
-  tag_extractors = Utility::createTagExtractors(bootstrap);
-  ASSERT_EQ(1, tag_extractors.size());
-  std::vector<Stats::Tag> tags;
-  std::string extracted_name =
-      tag_extractors.at(0)->extractTag("cluster.test_cluster.test_stat", tags);
-  EXPECT_EQ("cluster.test_stat", extracted_name);
-  ASSERT_EQ(1, tags.size());
-  EXPECT_EQ(tag_names.CLUSTER_NAME, tags.at(0).name_);
-  EXPECT_EQ("test_cluster", tags.at(0).value_);
-
-  // Add a custom regex for the name instead of relying on the default. The regex below just
-  // captures the entire string, and should override the default for this name.
-  custom_tag_extractor.set_regex("((.*))");
-  tag_extractors = Utility::createTagExtractors(bootstrap);
-  ASSERT_EQ(1, tag_extractors.size());
-  tags.clear();
-  // Use the same string as before to ensure the same regex is not applied.
-  extracted_name = tag_extractors.at(0)->extractTag("cluster.test_cluster.test_stat", tags);
-  EXPECT_EQ("", extracted_name);
-  ASSERT_EQ(1, tags.size());
-  EXPECT_EQ(tag_names.CLUSTER_NAME, tags.at(0).name_);
-  EXPECT_EQ("cluster.test_cluster.test_stat", tags.at(0).value_);
-
-  // Non-default custom name with regex should work the same as above
-  custom_tag_extractor.set_tag_name("test_extractor");
-  tag_extractors = Utility::createTagExtractors(bootstrap);
-  ASSERT_EQ(1, tag_extractors.size());
-  tags.clear();
-  // Use the same string as before to ensure the same regex is not applied.
-  extracted_name = tag_extractors.at(0)->extractTag("cluster.test_cluster.test_stat", tags);
-  EXPECT_EQ("", extracted_name);
-  ASSERT_EQ(1, tags.size());
-  EXPECT_EQ("test_extractor", tags.at(0).name_);
-  EXPECT_EQ("cluster.test_cluster.test_stat", tags.at(0).value_);
-
-  // Non-default custom name without regex should throw
-  custom_tag_extractor.set_regex("");
-  EXPECT_THROW_WITH_MESSAGE(
-      Utility::createTagExtractors(bootstrap), EnvoyException,
-      "No regex specified for tag specifier and no default regex for name: 'test_extractor'");
+  Utility::createTagProducer(bootstrap);
```

Comment:
This test seems to just exist now to show that this method doesn't throw.  Do you think we could add in a little logic to ensure the returned pointer is not `nullptr` and can be used (by maybe calling `produceTags()` on it or something)?

Reply:
Yeah, feeling I'm missing something here. +1 to add some tests against the returned value.